blaze-client/src/main/scala/org/http4s/blaze/client/PoolManager.scala

```diff
@@ -90,7 +93,7 @@ private final class PoolManager[F[_], A <: Connection[F]](
 
   private var isClosed = false
   private val curTotal = new AtomicInteger(0)
-  private val allocated = mutable.Map.empty[RequestKey, Int]
+  private val allocated = new ConcurrentHashMap[RequestKey, Int]()
```

Comment:
Initially, @satorg proposed to use `TrieMap`, but I think `ConcurrentHashMap` will be fine here. Besides the fact, `ConcurrentHashMap` [has a bit more performance](https://stackoverflow.com/a/38700133).

Reply:
Personally I am totally fine with the Java's `ConcurrentHashMap` too :)